freeipa/src/main/java/com/sequenceiq/freeipa/controller/DnsV1Controller.java

```diff
@@ -126,4 +128,44 @@ public void deleteDnsRecordsByFqdn(@ResourceCrn @NotEmpty String environmentCrn,
             throw new FreeIpaClientExceptionWrapper(e);
         }
     }
+
+    @Override
+    @CheckPermissionByRequestProperty(path = "environmentCrn", type = CRN, action = EDIT_ENVIRONMENT)
+    @Retryable(value = RetryableFreeIpaClientException.class,
+            maxAttemptsExpression = RetryableFreeIpaClientException.MAX_RETRIES_EXPRESSION,
+            backoff = @Backoff(delayExpression = RetryableFreeIpaClientException.DELAY_EXPRESSION,
+                    multiplierExpression = RetryableFreeIpaClientException.MULTIPLIER_EXPRESSION))
+    public void addDnsARecord(@RequestObject @Valid AddDnsARecordRequest request) {
+
+    }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = EDIT_ENVIRONMENT)
+    @Retryable(value = RetryableFreeIpaClientException.class,
+            maxAttemptsExpression = RetryableFreeIpaClientException.MAX_RETRIES_EXPRESSION,
+            backoff = @Backoff(delayExpression = RetryableFreeIpaClientException.DELAY_EXPRESSION,
+                    multiplierExpression = RetryableFreeIpaClientException.MULTIPLIER_EXPRESSION))
+    public void deleteDnsARecord(@ResourceCrn @NotEmpty String environmentCrn, @NotEmpty String dnsZone, @NotEmpty String record) {
+
+    }
+
+    @Override
+    @CheckPermissionByRequestProperty(path = "environmentCrn", type = CRN, action = EDIT_ENVIRONMENT)
+    @Retryable(value = RetryableFreeIpaClientException.class,
+            maxAttemptsExpression = RetryableFreeIpaClientException.MAX_RETRIES_EXPRESSION,
+            backoff = @Backoff(delayExpression = RetryableFreeIpaClientException.DELAY_EXPRESSION,
+                    multiplierExpression = RetryableFreeIpaClientException.MULTIPLIER_EXPRESSION))
+    public void addDnsCnameRecord(@RequestObject @Valid AddDnsCnameRecordRequest request) {
+
+    }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = EDIT_ENVIRONMENT)
+    @Retryable(value = RetryableFreeIpaClientException.class,
+            maxAttemptsExpression = RetryableFreeIpaClientException.MAX_RETRIES_EXPRESSION,
+            backoff = @Backoff(delayExpression = RetryableFreeIpaClientException.DELAY_EXPRESSION,
+                    multiplierExpression = RetryableFreeIpaClientException.MULTIPLIER_EXPRESSION))
+    public void deleteDnsCnameRecord(@ResourceCrn @NotEmpty String environmentCrn, @NotEmpty String dnsZone, @NotEmpty String record) {
+
+    }
```

Comment:
It looks like this commit just adds the API and not the implementation. Is that intended?

Reply:
yes, implementation is in progress

Reply:
This is just for reviewing the API for the different teams.